### DISCUSSION TITLE
Custom layout example for .NET Maui?

### DISCUSSION THREAD

**Discussion by JohnHDev** [dotnet/maui#8619] [Q&A]
We are looking at migrating a custom layout from XF to .NET Maui, but we can't find any documentation on how to do this, or to even create a layout from scratch.

The existing XF documentation is here:
https://docs.microsoft.com/en-us/xamarin/xamarin-forms/user-interface/layouts/custom

In our (so far brief) investigation we can see we need a layout manager that arranges the children, but I don't see a way to set the height of the layout once the children have been arranged. We can use HeightRequest but that isn't always working (it doesn't work at all if the children have been created via an ItemsSource binding for example).

Any pointers would be appreciated.

Thanks
John

**Comment by xtuzy** [comment]
overridearrange or overridemeasure have a return, it maybe decide layout size. you can see maui's layout source code.

**Comment by davidbritch** [comment]
https://github.com/hartez/CustomLayoutExamples

**Comment by MartyIX** [comment]
There is now:

* https://learn.microsoft.com/en-us/dotnet/maui/user-interface/layouts/custom?view=net-maui-8.0#create-a-custom-layout-type
* https://github.com/dotnet/maui/blob/main/docs/design/layout.md

**Comment by JTOne123** [comment]
I just want to leave my article here https://pavlodatsiuk.hashnode.dev/converting-xamarinforms-custom-layouts-to-maui, it's not fully related to the question, but it can help to quick start with the matter.
